Sources/NIOPosix/MultiThreadedEventLoopGroup.swift

```diff
@@ -460,21 +460,32 @@ internal struct ScheduledTask {
     ///     This means, the ids need to be unique for a given ``SelectableEventLoop`` and they need to be in ascending order.
     @usableFromInline
     let id: UInt64
-    let task: () -> Void
-    private let failFn: (Error) -> Void
+
     @usableFromInline
     internal let readyTime: NIODeadline
 
+    @usableFromInline
+    enum Kind {
+        case task(task: () -> Void, failFn: (Error) -> Void)
+        case callback(any NIOScheduledCallbackHandler)
+    }
+
+    @usableFromInline
+    let kind: Kind
+
+    // TODO: Should these be .init() or should they be static functions?
```

Comment:
I think `init` is fine, it's `internal` API anyway so doesn't matter too much.

Reply:
Yeah, it doesn't really matter I suppose, I've removed the TODO in 21fd1cca5.